### PR TITLE
Minor improvements to setup.py

### PR DESCRIPTION
**Pull request by agronholm**
I assume that as of at least v1.3, this is no longer in beta?

### CODE REVIEW

**Comment by njsmith** [comment]
> no longer in beta?
Heh, good catch.

File: setup.py [hunk 1]
```diff
@@ -6,15 +8,14 @@
     name="async_generator",
     version=__version__,
     description="Async generators for Python 3.5",
-    long_description=open("README.rst").read(),
+    long_description=Path(__file__).with_name("README.rst").read_text('utf-8'),
```
Comment (njsmith) [inline setup.py]: What does this do?

Comment (agronholm) [inline setup.py]: Figures out the path to `setup.py`, even if it's not in the current directory, and then changes the file name of the path to `README.rst` and finally reads the file contents as UTF-8 (the explicit encoding is necessary on Anaconda installs [like on RTD] where str.decode() for some incomprehensible reason does not default to utf-8!).
In the early days of my most popular project, APScheduler, I got a PR from someone who was doing RPM packaging of APScheduler. They were complaining that apscheduler's setup.py assumes that the working directory is the project directory, which apparently is not always the case. This line avoids that by not assuming anything about the CWD (and it closes the properly too, not that it really matters in the case of setup.py).

Comment (njsmith) [inline setup.py]: > assumes that the working directory is the project directory, which apparently is not always the case.

......wuuut. really? .....I'm suddenly paranoid about all my packages. really? yikes.

**Comment by njsmith** [comment]
Anything else you need, or should I roll a 1.3 release to get the `yield_` change out for you?

**Comment by agronholm** [comment]
I took a stab at adding type hints but I wasn't sure what types to put where, so I gave up for now. I've already released v1.3 of asyncio_extras which now depends on async_generator.

**Comment by njsmith** [comment]
'kay, v1.4 is up on pypi (not sure why I said 1.3 above)

**Pull request merged**